src/Microsoft.AspNet.Mvc.Core/Rendering/Html/HtmlHelper.cs

```diff
@@ -558,7 +589,7 @@ protected virtual MvcForm CreateForm()
             return new HtmlString(elements);
         }
 
-        protected virtual string GenerateDisplayName([NotNull] ModelMetadata metadata, string htmlFieldName)
+        protected virtual string GenerateDisplayName([NotNull] ModelMetadata metadata, string expression)
```

Comment:
Confused/curious: So this was an `htmlFieldName` before and now it's an expression? I see the code below showing how it's "kind of" an expression, seems odd going the full distance from a field name to an expression.